### PR TITLE
Travis-CI builds

### PR DESCRIPTION
**Pull request by awschristou**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Technical debt (change which improves the maintainability of the codebase, but does not change functionality)
- [ ] Testing (change which modifies or adds test coverage, but does not change functionality)
- [ ] Documentation (change which modifies or adds documentation, but does not change functionality)

## Description

Adding Travis-CI builds

## Related Issue(s)

#115 

## Testing

Watching the travis build output to see that it successfully builds and runs the tests

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+language: node_js
```
Comment (mpiroc) [inline .travis.yml]: Does travis automatically do a `npm install`?

Comment (awschristou) [reply .travis.yml]: No, I added that into the script

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+matrix:
+  include:
+    - os: linux
+    - os: osx
+
+before_install:
+  # Set up X Virtual Framebuffer because an instance of VS Code runs the tests
+  # See https://docs.travis-ci.com/user/gui-and-headless-browsers/#using-xvfb-to-run-tests-that-require-a-gui
+  - if [ $TRAVIS_OS_NAME == "linux" ]; then
```
Comment (mpiroc) [inline .travis.yml]: 1. What's the difference between this and `xvfb-run npm test --silent` from our `buildspec.yaml`?
2. Why isn't this needed for osx?

Comment (mpiroc) [reply .travis.yml]: Also, why isn't this needed for osx?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      export CXX="g++-4.9" CC="gcc-4.9" DISPLAY=:99.0;
+      sh -e /etc/init.d/xvfb start;
+      sleep 3;
+    fi
+
+script:
```
Comment (mpiroc) [inline .travis.yml]: Shouldn't these scripts be identical to the ones in `buildspec.yaml`?

**Pull request merged**